tests/bdd/features/cordon/node/test_cordon_node.py

```diff
@@ -0,0 +1,451 @@
+    )
+    yield
+    Docker.restart_container("core")
+    Cluster.cleanup()
+    remove_all_cordons(NODE_NAME_1)
+    remove_all_cordons(NODE_NAME_2)
+    delete_all_volumes()
+
+
+@pytest.fixture
+def tmp_files():
+    files = []
+    for index in range(0, 1):
+        files.append(f"/tmp/disk_{index}")
+    yield files
+
+
+@pytest.fixture
+def disks(tmp_files):
+    for disk in tmp_files:
+        if os.path.exists(disk):
+            os.remove(disk)
+        with open(disk, "w") as file:
+            file.truncate(100 * 1024 * 1024)
+    # /tmp is mapped into /host/tmp within the io-engine containers
+    yield list(map(lambda file: f"/host{file}", tmp_files))
+
+    for disk in tmp_files:
+        if os.path.exists(disk):
+            os.remove(disk)
+
+
+# Fixture used to pass the volume create request between test steps.
+@pytest.fixture(scope="function")
+def create_request():
+    return {}
+
+
+###### utility functions ######
+
+
+def cordon_node(node_name, label):
+    ApiClient.nodes_api().put_node_cordon(node_name, label)
+    assert is_cordoned(node_name) == True
+
+
+def create_volume(vol_uuid):
+    request = CreateVolumeBody(
+        VolumePolicy(False), NUM_VOLUME_REPLICAS, VOLUME_SIZE, False
+    )
+    ApiClient.volumes_api().put_volume(vol_uuid, request)
+    # Check that the volume was created.
+    volumes = ApiClient.volumes_api().get_volume(vol_uuid)
+
+
+def create_volume_and_fail(vol_uuid):
+    request = CreateVolumeBody(
+        VolumePolicy(False), NUM_VOLUME_REPLICAS, VOLUME_SIZE, False
+    )
+    try:
+        ApiClient.volumes_api().put_volume(vol_uuid, request)
+        assert False
+    except Exception as e:
+        print("failed, as expected")
+    # Check that the volume wasn't created.
+    try:
+        volumes = ApiClient.volumes_api().get_volume(vol_uuid)
+        assert False
+    except Exception as e:
+        print("no volume, as expected")
+
+
+def delete_volume(vol_uuid):
+    ApiClient.volumes_api().del_volume(vol_uuid)
+
+
+def delete_all_volumes():
+    volumes = ApiClient.volumes_api().get_volumes()
+    for vol in volumes.entries:
+        ApiClient.volumes_api().del_volume(vol.spec.uuid)
+
+
+def get_volumes():
+    return ApiClient.volumes_api().get_volumes()
+
+
+def is_cordoned(node_name):
+    node = ApiClient.nodes_api().get_node(node_name)
+    present = False
+    try:
+        print("%v", node.spec.cordondrainstate)
+        assert node.spec.cordondrainstate["cordonedstate"]["cordonlabels"] != []
+        return True
+    except AttributeError as e:
+        return False
+
+
+def remove_all_cordons(node_name):
+    if is_cordoned(node_name):
+        node = ApiClient.nodes_api().get_node(node_name)
+        for label in node.spec.cordondrainstate["cordonedstate"]["cordonlabels"]:
+            ApiClient.nodes_api().delete_node_cordon(node_name, label)
+
+
+def share_volume(vol_uuid):
+    ApiClient.volumes_api().put_volume_share(vol_uuid, "nvmf")
```

Comment:
Did you mean share rather than publish?
Also this doesn't seem used?

Reply:
This was created and got left behind after I was trying to implement the remaining tests. I'll remove it.